### ISSUE TITLE
Center window by default

### ISSUE THREAD

**Issue by emersion** [andlabs/libui#183]
Windows are not centered by default. Calling `Show` and `Center` does not work well because the window is first displayed in a corner of the screen and is then centered.

Provide a way to center the window by default.

**Comment by andlabs** [comment]
What OS? Center not working before Show is a bug.

**Comment by emersion** [comment]
Okay. It's happening with GTK:
```
(Desktop-Bridge:5073): Gdk-CRITICAL **: gdk_window_get_screen: assertion 'GDK_IS_WINDOW (window)' failed

(Desktop-Bridge:5073): Gdk-CRITICAL **: gdk_screen_get_monitor_at_window: assertion 'GDK_IS_SCREEN (screen)' failed

(Desktop-Bridge:5073): Gdk-CRITICAL **: gdk_screen_get_monitor_workarea: assertion 'GDK_IS_SCREEN (screen)' failed
```

**Comment by andlabs** [comment]
Yes, the GTK+ window moving is a giant mess right now due to having to deal with how Wayland works, and I'm not yet sure how to solve it.

**Comment by emersion** [comment]
On GTK and Windows the default position is not that bad, but on macOS it shows in at the bottom left corner. Is there a way to change this?

**Comment by emersion** [comment]
Looking at the code, why don't you just use `window.set_position(Gtk.WindowPosition.CENTER)`?

**Comment by andlabs** [comment]
I do not fully understand how that function works, namely I do not know if it will work *at any time*. I can ask about this easily, though.

**Comment by emersion** [comment]
Here: https://elementary.io/docs/code/getting-started#gtk-window-and-gtk-button

> Then we tell the window manager that we want to place this window in the center of the screen instead of in the default position (which is usually the top left).

I think you're right, `gtk_window_set_position` only works when the window is not shown, since the window manager will use it when opening the window. So the fix would be to call `gtk_window_set_position` when the window is hidden, and use your current code otherwise.
